pkgs/by-name/xr/xrock/package.nix

```diff
@@ -0,0 +1,36 @@
+{
+  fetchFromGitHub,
+  lib,
+  libusb1,
+  pkg-config,
+  stdenv,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "xrock";
+  version = "1.1.2";
+
+  src = fetchFromGitHub {
+    owner = "xboot";
+    repo = "xrock";
+    tag = "v${version}";
+    hash = "sha256-4b/Us0UybWL2XQOqwR36TAEUmbx3WOtKi6H9VYetweA=";
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ libusb1 ];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp xrock $out/bin
+  '';
```

Comment:
shouldn't be necessary. I think nix will install these in the right place for you.

Reply:
The build system runs `make install` which puts the binary into `/usr/local/bin`. Unfortunately the Make script does not respect variables like PREFIX. So I'm doing the install step myself. https://github.com/xboot/xrock/blob/master/Makefile#L70

Reply:
Forgot to add runhooks for pre- and post install phases and also mainProgram.